### PR TITLE
Fix device_label size

### PR DESCRIPTION
**Pull request by jojoquivogn**
https://github.com/Scille/parsec-cloud/issues/1937

### CODE REVIEW

File: parsec/api/protocol/types.py [hunk 1]
```diff
@@ -98,6 +98,22 @@ def new(cls: Type[DeviceIDTypeVar]) -> DeviceIDTypeVar:
...
+    def __new__(cls, device_label: str) -> "DeviceLabel":
+        device_label = normalize("NFC", device_label)
+
+        if not 0 < _bytes_size(device_label) < 255:
+            raise ValueError("Invalid device_label")
+
+        return super(DeviceLabel, cls).__new__(cls, device_label)
+
+
+class DeviceLabelField(fields.String):  # type: ignore
```
Comment (jojoquivogn) [inline parsec/api/protocol/types.py:111]: `# type: ignore` for avoid mypy strange error (only quick fix): 
```
parsec/api/protocol/types.py:111: error: Class cannot subclass 'String' (has type 'Any')  [misc]
parsec/api/protocol/types.py:111: error: Base type String becomes "Any" due to an unfollowed import  [no-any-unimported]
```

File: parsec/api/protocol/types.py [hunk 2]
```diff
@@ -98,6 +98,22 @@ def new(cls: Type[DeviceIDTypeVar]) -> DeviceIDTypeVar:
         return cls(f"{uuid4().hex}@{uuid4().hex}")
 
 
+class DeviceLabel(str):
+    def __new__(cls, device_label: str) -> "DeviceLabel":
+        device_label = normalize("NFC", device_label)
+
+        if not 0 < _bytes_size(device_label) < 255:
+            raise ValueError("Invalid device_label")
```
Comment (touilleMan) [inline parsec/api/protocol/types.py:106]: message should be `"Invalid device label"`

File: parsec/api/protocol/types.py (hunk 1 above)
Comment (touilleMan) [reply parsec/api/protocol/types.py:111]: why not using `str_based_field_factory` ?

```python
DeviceLabelField = fields.str_based_field_factory(DeviceLabel)
```

**Review by touilleMan — APPROVED** [review]

**Review by touilleMan — APPROVED** [review]
image

**Pull request merged**